app/src/main/java/com/example/foodie/common/ui/SharedViewModel.kt

```diff
@@ -148,21 +151,18 @@ class SharedViewModel @Inject constructor(
         viewModelScope.launch {
             Log.d("SharedViewModel", "Saving filter with MealType: $mealType, Time: $time")
 
-            // Используем async для параллельного выполнения сохранений
-            val saveTimeDeferred = async { dataPreference.saveTime(time) }
-            val saveMealTypeDeferred = async { dataPreference.saveMealType(mealType) }
-
-            // Ожидаем завершения выполнения обеих операций
-            saveTimeDeferred.await()
-            saveMealTypeDeferred.await()
+            // Параллельное выполнение сохранений без ожидания результата
+            launch { dataPreference.saveTime(time) }
+            launch { dataPreference.saveMealType(mealType) }
 
             // Обновляем значения после завершения операций
             _lastTime.value = time
```

Comment:
update values will occur before the execution of coroutine above
use job to wait 
job = launch{}
job.join()